sa/sa.go

```diff
@@ -400,38 +409,39 @@ func (ssa *SQLStorageAuthority) GetCertificate(ctx context.Context, serial strin
 		return core.Certificate{}, err
 	}
 
-	certObj, err := ssa.dbMap.Get(core.Certificate{}, serial)
-	if err != nil {
-		return core.Certificate{}, err
-	}
-	if certObj == nil {
-		ssa.log.Debug(fmt.Sprintf("Nil cert for %s", serial))
+	var cert core.Certificate
+	err := ssa.dbMap.SelectOne(&cert, fmt.Sprintf("SELECT %s FROM certificates WHERE serial = %q", CertificateFields, serial))
+	if err == sql.ErrNoRows {
 		return core.Certificate{}, core.NotFoundError(fmt.Sprintf("No certificate found for %s", serial))
 	}
-
-	certPtr, ok := certObj.(*core.Certificate)
-	if !ok {
-		ssa.log.Debug("Failed to convert cert")
-		return core.Certificate{}, fmt.Errorf("Error converting certificate response for %s", serial)
+	if err != nil {
+		return core.Certificate{}, err
 	}
-	return *certPtr, err
+	return cert, err
 }
 
 // GetCertificateStatus takes a hexadecimal string representing the full 128-bit serial
 // number of a certificate and returns data about that certificate's current
 // validity.
-func (ssa *SQLStorageAuthority) GetCertificateStatus(ctx context.Context, serial string) (status core.CertificateStatus, err error) {
+func (ssa *SQLStorageAuthority) GetCertificateStatus(ctx context.Context, serial string) (core.CertificateStatus, error) {
 	if !core.ValidSerial(serial) {
 		err := fmt.Errorf("Invalid certificate serial %s", serial)
 		return core.CertificateStatus{}, err
 	}
 
-	certificateStats, err := ssa.dbMap.Get(core.CertificateStatus{}, serial)
+	var status core.CertificateStatus
+	err := ssa.dbMap.SelectOne(&status, fmt.Sprintf(
+		"SELECT %s FROM certificateStatus WHERE serial = %q",
```

Comment:
Definitely let's not DIY our SQL interpolation like this. Use Gorp's interpolation technique.